### DISCUSSION TITLE
How can I open the scrollbar in the reader

### DISCUSSION THREAD

**Discussion by jiijkkk** [spacecowboy/Feeder#145] [Q&A] [answered]
Such as the title， I cannot find the scrollbar so I don't know the length of the artical. Is there a scrollbar in the reader ?

**Comment by spacecowboy** [comment]
No. There is no scrollbar in Jetpack Compose to show.

**Reply by s38b35M5** [reply]
I came here to create a FR for scroll position indicator. It's useful for an indication of article length, feed length, relative position, etc.

There's no way to accomplish this for feeder @spacecowboy ?

**Reply by spacecowboy** [reply]
There is no built in scrollbar as I said.

And building one is not simple.

**Comment by s38b35M5** [comment]
I trust that it's harder than these pages [1](https://stackoverflow.com/questions/71838279/how-to-make-a-scrollbar-view-with-jetpack-compose) [2](https://medium.com/@ajitesh.gupta56/scroll-bars-in-jetpack-compose-android-ff787f9b522) make it seem. I wish I could wrap my ADHD brain around coding to help ...

**Reply by spacecowboy** [reply]
Thanks for the links! I'll take another look at it!

**Reply by spacecowboy** [reply]
But looking at it, neither one uses `LazyColumn`, which is part of the challenge.

**Reply by shuvashish76** [reply]
@spacecowboy Have a look on this [fastscroll](https://github.com/zhanghai/AndroidFastScroll) library.

**Reply by spacecowboy** [reply]
I realise you want to help but please only submit stuff which you understand is compatible.

It says as the first line: Fast scroll for Android RecyclerView and more.

Feeer is built using Jetpack Compose and not the old Views-system.

**Reply by s38b35M5** [reply]
I'm an old tech nerd with only python and .net programming experience. I hope to get to a point where I can help with this problem, but for now, android-studio makes no sense when I load the repo. I'll have to take my time. TY for an awesome app!

**Reply by spacecowboy** [reply]
remember to read the readme and clone with recursive enabled

**Reply by shuvashish76** [reply]
> I realise you want to help but please only submit stuff which you understand is compatible.

Sorry I'm not a dev. I noticed some Jetpack Compose based projects use https://github.com/nanihadesuka/LazyColumnScrollbar please have a look on it. Thanks

**Reply by spacecowboy** [reply]
> > I realise you want to help but please only submit stuff which you understand is compatible.
> 
> Sorry I'm not a dev. I noticed some Jetpack Compose based projects use https://github.com/nanihadesuka/LazyColumnScrollbar please have a look on it. Thanks

That might be something that works! Thanks for the tip

**Reply by spacecowboy** [reply]
There are some issues, but I've asked the maintainer if there is a setting I can tweak.

[ScrollBarDefault2.webm](https://github.com/user-attachments/assets/408bacec-f8c6-4a30-90d0-401f7f671fe0)
[ScrollBarFixedLength2.webm](https://github.com/user-attachments/assets/1c3b154c-cce4-4bab-8a1b-db1d9b152ff0)

**Reply by shuvashish76** [reply]
If that library doesn't work try this : https://github.com/reddit/IndicatorFastScroll popular library by _reddit_ but looks abandoned

**Reply by spacecowboy** [reply]
> If that library doesn't work try this : https://github.com/reddit/IndicatorFastScroll popular library by _reddit_ but looks abandoned

No. It says plainly on the first page why that can't work with Feeder:

> Works with standard RecyclerViews and in any layout

Feeder is built with Jetpack Compose. Not the legacy view layout system. It's a completely different way to build the UI.

**Comment by s38b35M5** [comment]
Pretty out of my element here, but maybe I'm on the right path. I just need a few weeks to familiarize with a new IDE and Kotlin...

In the meantime, here is what I am looking at:
```
import androidx.compose.foundation.VerticalScrollbar
import androidx.compose.foundation.rememberScrollbarAdapter
```

```
val scrollBarAdapter = rememberScrollbarAdapter(articleListState)
        VerticalScrollbar(
            modifier = Modifier.align(Alignment.CenterEnd).fillMaxHeight(),
            adapter = scrollBarAdapter
        )
```
Apologies for my fumbling. I used to understand programming. I let it get away from me...

**Comment by s38b35M5** [comment]
>Uh-huh. Did you ChatGPT that?
>Because question: How do you connect it to the LazyColumns: ScrollState`?

Yes, ChatGPT (Or Gemini, I forget) I asked for an example of how to integrate a scroll indicator. It initially provided ScrollState, but I just am way out of my depth here. Disregard me. Apologies.

I am working to wrap my head around AS IDE and there's a steep curve. I have feeder working on the VHD, so I can at least play with your code and see where it goes. Enjoy your weekend...

**Reply by spacecowboy** [reply]
So there is this: https://developer.android.com/reference/kotlin/androidx/compose/foundation/package-summary#VerticalScrollbar(androidx.compose.foundation.ScrollbarAdapter,androidx.compose.ui.Modifier,kotlin.Boolean,androidx.compose.foundation.ScrollbarStyle,androidx.compose.foundation.interaction.MutableInteractionSource)

But that is not something which is actually importable.

Sorry for my rudeness earlier. But my experience is that ChatGPT is quite shit at Jetpack Compose, basicaly because there isn't a lot of open source to train on / steal from yet.

**Reply by s38b35M5** [reply]
I'll lay low and learn, and maybe we can touch base when I have more meaningful input. No worries about rudeness. This is your app. I'm just trying to learn on probably the wrong project and methods.

**Comment by spacecowboy** [comment, marked as answer]
https://github.com/spacecowboy/Feeder/pull/350

**Reply by s38b35M5** [reply]
Terrific work!
